easybuild/framework/easyblock.py

```diff
@@ -2150,6 +2150,22 @@ def prepare_step(self, start_dir=True, load_tc_deps_modules=True):
             self.log.info("Loading extra modules: %s", extra_modules)
             self.modules_tool.load(extra_modules)
 
+        # Setup CUDA cache if required. If we don't do this, CUDA will use the $HOME for its cache files
+        if get_software_root('CUDA') or get_software_root('CUDAcore'):
```

Comment:
@Flamefire This belongs in a separate function imho, which we just call here.

How about a `set_up_cuda_cache` function in filetools?

Reply:
Locations considered for that function:
- filetools: This is not reading/writing of files but simply changing some env vars
- environment: That contains only the most basic functions
- toolchain: Likely very good fit but excludes allow_system_deps and extra_modules options if done in `prepare` and needs access to EasyBlock.builddir for the default
- EasyBlock: Easiest and also makes sense. Could make it private if we want to decide on the exact location later